src/core/sys/solaris/sys/elf.d

```diff
@@ -363,165 +54,28 @@ enum SHT_SUNW_SIGNATURE = 0x6ffffff6;
 enum SHT_SUNW_DEBUG     = 0x6ffffff9;
-enum SHT_SUNW_move      = 0x6ffffffa;
-enum SHT_SUNW_COMDAT    = 0x6ffffffb;
-enum SHT_SUNW_syminfo   = 0x6ffffffc;
 enum SHT_SUNW_verdef    = 0x6ffffffd;
-enum SHT_GNU_verdef     = SHT_SUNW_verdef;
 enum SHT_SUNW_verneed   = 0x6ffffffe;
-enum SHT_GNU_verneed    = SHT_SUNW_verneed;
 enum SHT_SUNW_versym    = 0x6fffffff;
-enum SHT_GNU_versym     = SHT_SUNW_versym;
-enum SHT_HISUNW         = 0x6fffffff;
-enum SHT_HIOS           = 0x6fffffff;
-enum SHT_GNU_ATTRIBUTES = 0x6ffffff5;
-enum SHT_GNU_HASH       = 0x6ffffff6;
-enum SHT_GNU_LIBLIST    = 0x6ffffff7;
-enum SHT_CHECKSUM       = 0x6ffffff8;
-enum SHT_LOPROC         = 0x70000000;
-enum SHT_HIPROC         = 0x7fffffff;
-enum SHT_LOUSER         = 0x80000000;
-enum SHT_HIUSER         = 0xffffffff;
-
-enum SHF_WRITE            = 0x01;
-enum SHF_ALLOC            = 0x02;
-enum SHF_EXECINSTR        = 0x04;
-enum SHF_MERGE            = 0x10;
-enum SHF_STRINGS          = 0x20;
-enum SHF_INFO_LINK        = 0x40;
-enum SHF_LINK_ORDER       = 0x80;
-enum SHF_OS_NONCONFORMING = 0x100;
-enum SHF_GROUP            = 0x200;
-enum SHF_TLS              = 0x400;
-enum SHF_COMPRESSED       = 0x800;
-enum SHF_MASKOS = 0x0ff00000;
-
-enum SHF_MASKPROC = 0xf0000000;
-
-enum SHN_UNDEF       = 0;
-enum SHN_LORESERVE   = 0xff00;
-enum SHN_LOPROC      = 0xff00;
-enum SHN_HIPROC      = 0xff1f;
-enum SHN_LOOS        = 0xff20;
-enum SHN_LOSUNW      = 0xff3f;
-enum SHN_SUNW_IGNORE = 0xff3f;
-enum SHN_HISUNW      = 0xff3f;
-enum SHN_HIOS        = 0xff3f;
-enum SHN_ABS         = 0xfff1;
-enum SHN_COMMON      = 0xfff2;
-enum SHN_XINDEX      = 0xffff;
-enum SHN_HIRESERVE   = 0xffff;
-
-struct Elf32_Sym
-{
-    Elf32_Word    st_name;
-    Elf32_Addr    st_value;
-    Elf32_Word    st_size;
-    ubyte         st_info;
-    ubyte         st_other;
-    Elf32_Half    st_shndx;
-}
-
-struct Elf64_Sym
-{
-    Elf64_Word    st_name;
-    ubyte         st_info;
-    ubyte         st_other;
-    Elf64_Half    st_shndx;
-    Elf64_Addr    st_value;
-    Elf64_Xword   st_size;
-}
-
-enum STN_UNDEF  = 0;
-
-extern (D)
-{
-    auto ELF32_ST_BIND(T)(T val) { return cast(ubyte)val >> 4; }
-    auto ELF32_ST_TYPE(T)(T val) { return val & 0xf; }
-    auto ELF32_ST_INFO(B, T)(B bind, T type) { return (bind << 4) + (type & 0xf); }
-    alias ELF32_ST_BIND ELF64_ST_BIND;
-    alias ELF32_ST_TYPE ELF64_ST_TYPE;
-    alias ELF32_ST_INFO ELF64_ST_INFO;
-}
-
-enum STB_LOCAL  = 0;
-enum STB_GLOBAL = 1;
-enum STB_WEAK   = 2;
-enum STB_NUM    = 3;
-enum STB_LOPROC = 13;
-enum STB_HIPROC = 15;
-
-enum STT_NOTYPE  = 0;
-enum STT_OBJECT  = 1;
-enum STT_FUNC    = 2;
-enum STT_SECTION = 3;
-enum STT_FILE    = 4;
-enum STT_COMMON  = 5;
-enum STT_TLS     = 6;
-enum STT_NUM     = 7;
-enum STT_LOOS    = 10;
-enum STT_HIOS    = 12;
-enum STT_LOPROC  = 13;
-enum STT_HIPROC  = 15;
 
 extern (D)
 {
     auto ELF32_ST_VISIBILITY(O)(O o) { return o & 0x07; }
     alias ELF32_ST_VISIBILITY ELF64_ST_VISIBILITY;
 }
```

Comment:
For reviewers:
Solaris defines `ELF32_ST_VISIBILITY` differently to all other Linux/BSDs (where it's `return o & 0x03`).